db/migrate/20190123145942_add_type_column_to_service_order.rb

```diff
@@ -0,0 +1,15 @@
+class AddTypeColumnToServiceOrder < ActiveRecord::Migration[5.0]
+  def up
+    add_column :service_order, :type, :string
+
+    say_with_time("Set service_order type") do
+      ServiceOrder.all.each do |service_order|
+        service_order.update_attribute(:type, service_order.miq_requests.pluck(:type))
```

Comment:
So 2 things

1. Type is reserved for STI to just taking the type from MiqRequest is going to be an issue.  If you're going to set a `:type` column it should match a sub-class of `ServiceOrder` so something like if MiqRequest is `ServiceTemplateProvisionRequest` then ServiceOrder would become `ServiceProvisionOrder` and `ServiceTemplateTransformationPlanRequest` => `ServiceTransformationOrder`

2. Updating for every service_order isn't very efficient, it'd be much better to group_by miq_request type and do an update_all

Reply:
> service_order.miq_requests.pluck(:type)

That yields an array of all the types, right?

Possibly something like

```
types = service_order.miq_requests.pluck(:type).sort.uniq
service_order.update_attribute(:type, types.first) if types.count == 1
```

?

Reply:
I think..

```sql
SELECT string_agg(DISTINCT type, ', ') AS type, service_order_id FROM miq_requests WHERE service_order_id IS NOT NULL GROUP BY service_order_id;
```

gives you the right list, so you can transform it into a single update, and it should mostly work.

We may want to clear any types with a comma after though.

Reply:
@agrare Ok, we don't want STI here, so I've changed the column to `request_type` as per @himdel's suggestion. It's now a sorted and comma-joined string in the event there is more than one request type.

If there's a more efficient way of doing this update I think I will need a little help putting it together.